src/tiled/wangbrush.cpp

```diff
@@ -544,7 +544,7 @@ void WangBrush::updateBrush()
                 continue;
 
             QPoint p = adjacentPositions[i];
-            if (!currentLayer->contains(p) || currentLayer->cellAt(p).isEmpty())
+            if ((!currentLayer->contains(p) && !mapDocument()->map()->infinite()) || currentLayer->cellAt(p).isEmpty())
```

Comment:
You can drop the `(!currentLayer->contains(p) && !mapDocument()->map()->infinite()) ||` part now, since just `currentLayer->cellAt(p).isEmpty()` will work in both cases.